spec/spec_helper_local.rb

```diff
@@ -13,6 +16,22 @@
     Facter.clear_messages
 
     RSpec::Mocks.setup
+
+    test_keys = OpenSSL::PKey::RSA.new(2049)
+    test_cert = OpenSSL::X509::Certificate.new
+    test_cert.version = 2
+    test_cert.serial = 1
+    test_cert.subject = OpenSSL::X509::Name.parse('/CN=test-cert.example.org')
+    test_cert.public_key = test_keys.public_key
+    test_cert.not_before = Time.now
+    test_cert.not_after = test_cert.not_before + 3600 # 1 hour
+    test_cert.sign(test_keys, OpenSSL::Digest::SHA256.new)
+
+    resp_header = { 'Content-Type': 'application/x-x509-ca-cert' }
+    stub_request(:get, 'http://example.org/cert.der')
+      .to_return(status: 200, body: test_cert.to_der, headers: resp_header)
+    stub_request(:get, 'http://example.org/cert.pem')
+      .to_return(status: 200, body: test_cert.to_pem, headers: resp_header)
```

Comment:
This whole block seems to only be used in `spec/unit/puppet/provider/cert_file/posix_spec.rb`

Could you move it there?